src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -4067,8 +4067,8 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                             BuildMethodTableThrowException(IDS_CLASSLOAD_BADFORMAT);
                         }
 
-                        if (IsStrLongerThan((char *)pszClassName, MAX_CLASS_NAME)
-                            || IsStrLongerThan((char *)pszNameSpace, MAX_CLASS_NAME)
+                        if (strnlen((char *)pszClassName, MAX_CLASS_NAME) >= MAX_CLASS_NAME
+                            || strnlen((char *)pszNameSpace, MAX_CLASS_NAME) >= MAX_CLASS_NAME
                             || (strlen(pszClassName) + strlen(pszNameSpace) + 1 >= MAX_CLASS_NAME))
```

Comment:
```suggestion
                        if (strnlen(pszClassName, MAX_CLASS_NAME) + strnlen(pszNameSpace, MAX_CLASS_NAME) + 1 >= MAX_CLASS_NAME)
```

Reply:
I do not see a reason for this check. It looks like a forgotten left-over. There is one other place in this file that calls `GetNameOfTypeRef` + `FindTypeDef` that does not validate the length.

I think even better fix would be to delete this check and `BFA_TYPEREG_NAME_TOO_LONG` error message.